### PR TITLE
Add env vars for MECA SFTP connection

### PR DESCRIPTION
**Pull request by diversemix**
See https://github.com/elifesciences/elife-xpub/issues/728

### CODE REVIEW

File: salt/elife-xpub/init.sls [hunk 1]
```diff
@@ -41,6 +41,11 @@ elife-xpub-environment-variables-for-configuration:
             export PUBSWEET_SECRET={{ pillar.elife_xpub.pubsweet.secret }}
             export S3_BUCKET={{ pillar.elife_xpub.s3.bucket }}
             export NODE_CONFIG_ENV={{ pillar.elife_xpub.deployment_target }}
+            export MECA_SFTP_HOST={{ pillar.elife_xpub.meca.sftp.connectionOptions.host }}
```
Comment (giorgiosironi) [inline salt/elife-xpub/init.sls]: some dummy values should be added to https://github.com/elifesciences/elife-xpub-formula/blob/master/salt/pillar/elife-xpub.sls as these will be copied into the AWS default

Comment (giorgiosironi) [reply salt/elife-xpub/init.sls]: `connectionOptions` goes against Salt convention to use snake case for pillar names `connection_options`; I don't think it's forbidden but non-standard

Comment (diversemix) [reply salt/elife-xpub/init.sls]: Ok ... happy to change this :+1:

File: salt/pillar/elife-xpub.sls [hunk 2]
```diff
@@ -21,6 +21,13 @@ elife_xpub:
     s3:
         bucket: fake_bucket
     meca:
+        sftp:
+            connection:
+                host: fake_host
+                port: fake_port
+                username: fake_username
+                password: fake_password
+            remotePath: fake_path
```
Comment (giorgiosironi) [inline salt/pillar/elife-xpub.sls]: also `remote_path` here?

**Comment by giorgiosironi** [comment]
Deploying in https://alfred.elifesciences.org/job/test-elife-xpub/347/console

**Pull request merged**